triton-vm/src/proof_item.rs

```diff
@@ -24,16 +112,48 @@ pub enum ProofItem {
     RevealedCombinationElement(XFieldElement),
     RevealedCombinationElements(Vec<XFieldElement>),
     FriCodeword(Vec<XFieldElement>),
-    FriProof(FriProof),
+    FriResponse(FriResponse),
     PaddedHeight(BFieldElement),
+    Uncast(Vec<BFieldElement>),
 }
 
-impl ProofItem {
+impl MayBeUncast for ProofItem {
+    fn uncast(&self) -> Vec<BFieldElement> {
+        if let Self::Uncast(vector) = self {
+            let mut str = vec![];
+            str.push(BFieldElement::new(vector.len().try_into().unwrap()));
+            str.append(&mut vector.clone());
+            str
+        } else {
+            self.encode()
+        }
+    }
+}
+
+impl ProofItem
+where
+    AuthenticationStructure<Digest>: BFieldCodec,
+    Vec<Vec<BFieldElement>>: BFieldCodec,
+    Vec<Vec<XFieldElement>>: BFieldCodec,
+    Digest: BFieldCodec,
+    Vec<BFieldElement>: BFieldCodec,
+    Vec<XFieldElement>: BFieldCodec,
+    Vec<Digest>: BFieldCodec,
+    BFieldElement: BFieldCodec,
+    XFieldElement: BFieldCodec,
+    FriResponse: BFieldCodec,
+{
     pub fn as_compressed_authentication_paths(
         &self,
     ) -> Result<AuthenticationStructure<Digest>, Box<dyn std::error::Error>> {
         match self {
             Self::CompressedAuthenticationPaths(caps) => Ok(caps.to_owned()),
+            Self::Uncast(str) => match AuthenticationStructure::<Digest>::decode(str) {
+                Ok(boxed_auth_struct) => Ok(*boxed_auth_struct),
+                Err(_) => Err(ProofStreamError::boxed(
+                    "cast to authentication structure failed",
+                )),
+            },
```

Comment:
From what I can see, you're throwing away the error message returned from the deeper call level here. The error produced by the file `bfield_codec.rs`. Maybe you would want to print that error as well? It should be printable here, I think.